Refresh `README.md`

Lamparter opened the pull request:
Some small documentation changes to make the README file more _readable_, including adding a rounded hero gif, a prettier features section structure, a refreshed licence section and (controversially) added emoji to the README file. Let me know what you think!

Lamparter committed:
Add pretty hero image

Lamparter committed:
Remove "learn more" section

Lamparter committed:
Change features section structure

Lamparter committed:
Add link to `libera.chat`

Lamparter committed:
Refresh license section

Lamparter committed:
Add emoji to `README.md`

(controversial, though it does give much better readability)

Lamparter committed:
Use standard copyright symbol

Lamparter committed:
Rename "development" section to "building from source"

Lamparter committed:
Add codebase structure to `README.md`

Lamparter commented:
@mjankowski lmao

![image](https://github.com/user-attachments/assets/9a9ddd69-84ae-4801-b0cf-8f87b5d147a3)

vmstan commented on README.md:
```diff
@@ -74,9 +60,27 @@ Mastodon acts as an OAuth2 provider, so 3rd party apps can use the REST and Stre
+├──log                               // Hosting logs (Host-specific backend; reserved for Mastodon hosting)
+├──public                            // Static resource files
+├──spec                              // Helpers, presenters, and other code models
+├──streaming                         // (Host-specific backend; reserved for Mastodon hosting)
```
This is the streaming API application which is Node.js based.

Lamparter replied:
I wrote this about two years ago 😅

Lamparter commented:
Thanks! 
If you want, I'll remove it, but I thought it might look good.

mjankowski commented:
Honestly when it comes to an excessive amount of emoji I'm of two minds.

There's part of me that I'd like to think is the sort of person to wear suits and sit in rooms with leather bound books and mahogany  furniture and smoke cigars while discussing serious business, and that part of me is frankly disgusted by things like this.

On the other hand, I once put an actual hamburger emoji in place of some horizontal lines in a "hamburger menu", and then just yesterday I put some sparkle emojis INTO MY PR TITLE here https://github.com/mastodon/mastodon/pull/32335

All that is to say, thank you for your service, and best wishes on your README updates.

renchap commented:
@andypiper do you want to have a look at this?

Lamparter committed:
Merge branch 'main' into docs

Lamparter committed:
Update codebase structure docs

Lamparter committed:
Remove emoji

I'm actually beginning to agree with the person who wears suits and sits in rooms with leather bound books and mahogany furniture and smokes cigars while discussing serious business.

mjankowski commented on README.md:
```diff
@@ -74,7 +60,25 @@ Mastodon acts as an OAuth2 provider, so 3rd party apps can use the REST and Stre
 The repository includes deployment configurations for **Docker and docker-compose** as well as specific platforms like **Heroku**, and **Scalingo**. For Helm charts, reference the [mastodon/chart repository](https://github.com/mastodon/chart). The [**standalone** installation guide](https://docs.joinmastodon.org/admin/install/) is available in the documentation.
 
-## Development
+### Codebase structure
```
I'm not sure this entire section is useful. Some of the descriptions here are either wrong, or sort of correct but phrased a little oddly.

Lamparter replied:
Better?

```diff
  ├──.devcontainer                     // Scripts and preparation files for running Mastodon in a development container such as github.dev
  ├──.github                           // Health files for GitHub such as issue templates
- ├──app                               // Collection of code files relating to the app's frontend
+ ├──app                               // The app frontend
  ├──bin                               // Scripts for running and hosting Mastodon
  ├──config                            // Code files relating to federated hosting and Ruby bootstrappers
  ├──db                                // Database files for Mastodon's information database (Host-specific backend)
- ├──dist                              // Collection of code files relating app hosting on domains and port relations
+ ├──dist                              // systemd services for running and hosting Mastodon and using nginx
- ├──lib                               // Code and file library
+ ├──lib                               // Library for functions used across the codebase
  ├──log                               // Hosting logs
  ├──public                            // Static resources
- ├──spec                              // Helpers, presenters, and other code models
+ ├──spec                              // Helpers, presenters, services, and other backend models
  ├──streaming                         // Streaming API service
  └──vendor
```

mjankowski replied:
High level - I'd probably delete this or add it to a developer-specific doc file, not the top level README.

More specifically:

- `app` has basically all the files (frontend, backend, activitypub, rest api, the core web app, db models, etc) - not just the frontend
- `bin` has mostly developer-useful tools, and only sort of hosting scripts
- `config` has a grab bag of stuff, not just federated hosting
- `spec` has the test suite
- Some of the others -- .github, log, public, streaming, db -- your descriptions are fine, but also sort of superfluous because the directory is pretty self explanatory and hopefully not confusing to anyone who needs to understand it?

Lamparter replied:
Perhaps better to just remove it entirely.

andypiper replied:
I think it probably makes sense to remove this section. Rails developers would likely be familiar with the basic structure.

Lamparter replied:
Done! Being a .NET developer probably doesn't help with something like this 😁

mjankowski commented:
I see you've removed many of the emojis. Given my previous comments on their inclusion I have very mixed feelings on this change. They did bring a certain childlike sense of  awe and wonder to this README (in a fun way!), but then also they made me wanted to delete the entire file. Sort of a mixed bag.

May I suggest that for basically all the other changes -- to the extent you have things here that are essentially very lightweight formatting changes, markdown style changes, etc ... and then other things which are slightly more meaningful copy/information changes ... you may have more success getting anyone with commit access to look at this if you break those up into smaller PRs with one narrow type of change per PR (and include before/after images, if relevant, if doing anything vaguely layout/design related).

Lamparter commented on README.md:
```diff
@@ -60,7 +60,7 @@ Mastodon is a **free, open-source social network server** based on ActivityPub w
 The repository includes deployment configurations for **Docker and docker-compose** as well as specific platforms like **Heroku**, and **Scalingo**. For Helm charts, reference the [mastodon/chart repository](https://github.com/mastodon/chart). The [**standalone** installation guide](https://docs.joinmastodon.org/admin/install/) is available in the documentation.
 
-## 🛠️ Development
+## 🛠️ Building from source
```
Is the previous "Development" good enough? I've gone ahead and reverted it.

Lamparter committed:
Revert rename of "development" section

andypiper commented on README.md:
```diff
@@ -1,24 +1,20 @@
-[releases]: https://github.com/mastodon/mastodon/releases
-[crowdin]: https://crowdin.com/project/mastodon
+<p align="center">
+  <a style="text-decoration:none" href="https://www.youtube.com/watch?v=IPSbNdBmWKE">
```
I like the softer rounded hero image here but it's a shame that we have lost the explanatory text that you can click through to see a video. Maybe add this underneath the image?

Lamparter replied:
Is it better now?

andypiper replied:
thanks!

Lamparter committed:
Remove codebase structure

andypiper reviewed (changes requested):
My suggestion would be to start a new PR (to avoid repeated force pushes / back and forth on changes here. I'm open to the general "nicer layout" items here!

Lamparter committed:
Merge branch 'mastodon:main' into docs

Lamparter committed:
Add back "watch video" tip

mjankowski commented on README.md:
````diff
@@ -138,17 +128,23 @@ Mastodon is **free, open-source software** licensed under **AGPLv3**.
+
+Licensed under GNU Affero General Public License as stated in the [LICENSE](LICENSE):
+
+```
````
If we're going to convert this to a pre/code block - can you hard wrap the contents at 80 cols?

Lamparter replied:
It won't look as good but sure

Lamparter committed:
Hard wrap license information to 80 columns

andypiper commented:
@renchap I do not have permissions to merge, but I'm OK with this in the current state.
